### ISSUE TITLE
new a11y rule: accessible tables

### ISSUE THREAD

**Issue by HamletDRC** [microsoft/tslint-microsoft-contrib#272]
1. tables should have either a caption or a title: https://www.w3.org/TR/WCAG10-HTML-TECHS/#table-summary-info
2. tables should have a summary attribute: https://www.w3.org/TR/WCAG10-HTML-TECHS/#table-summary-info, http://oaa-accessibility.org/wcag20/rule/3/
3. Provide terse substitutes for header labels with the "abbr" attribute on TH
4. Tables must have th element (http://oaa-accessibility.org/rule/4/', https://github.com/globant-ui/arialinter/blob/master/lib/rules/adaptable.js)
5. Summary attribute content must be unique (http://oaa-accessibility.org/rule/5/, https://github.com/globant-ui/arialinter/blob/master/lib/rules/adaptable.js)
6. Complex data tables must have ids on th elements. (http://oaa-accessibility.org/wcag20/rule/6/)
7.  For complex data tables table ids must be unique. (http://oaa-accessibility.org/wcag20/rule/7/)
8. Complex data table td elements must have header attributes. (http://oaa-accessibility.org/wcag20/rule/8/)
9. Complex data tables header ids must be on the page. (http://oaa-accessibility.org/wcag20/rule/9/)
10. Enforce scope prop is only used on `<th>` elements. From 
    scope.md

From HTML Code Sniffer: 
11. Not all th elements in this table have a scope attribute. These cells should contain a scope attribute to identify their association with td elements. 
    • http://www.w3.org/TR/WCAG20-TECHS/H63
12. If this table is a data table, and both a summary attribute and a caption element are present, the summary should not duplicate the caption.
    • http://www.w3.org/TR/WCAG20-TECHS/H39
    • http://www.w3.org/TR/WCAG20-TECHS/H73
13. Incorrect headers attribute on this td element. Expected "[Expected list of IDs]" but found "[Actual list of IDs]".
    • https://squizlabs.github.io/HTML_CodeSniffer/Standards/Section508/

presentational `<table>` elements must not use `<th>`, `<caption>` elements or the summary attribute. axe-core layout-table - wcag2a, wcag131

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated**; therefore, it and tslint-microsoft-contrib are no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint or tslint-microsoft-contrib locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_
